src/sentry/models/avatars/base.py

```diff
@@ -141,11 +150,12 @@ def save_avatar(cls, relation, type, avatar=None, filename=None, color=None):
                 file.delete()
 
             if photo:
-                setattr(instance, cls.file_fk(), photo.id)
+                if instance.file_fk() != instance.file_write_fk():
+                    setattr(instance, instance.file_fk(), None)
```

Comment:
What's the use case for having a different `file_write_fk`?

Reply:
I wanted to have all future avatar writes go to the silo paired file models. In single-tenant and self-hosted, user and sentryapp avatar writes still go to `File` because those environments lack the control silo bucket configuration options. 

This condition covers the scenario where a non-saas environment updates an avatar that was previously stored in `File` will now be written to `ControlFile` instead. Blanking the `file_id` column felt like good housekeeping as it would otherwise refer to a stale value.